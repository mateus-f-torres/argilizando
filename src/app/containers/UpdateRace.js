import {connect} from 'react-redux';
import {withRouter} from 'react-router-dom';
import {displayRace, lockRace} from '../actions';
import PlayerRace from '../components/PlayerRace.jsx';

const mapStateToProps = (state) => {
  return {
    race: state.race.chosen,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    getRace: (race) => {
      dispatch(displayRace(race));
    },
    lockRace: (race) => {
      dispatch(lockRace(race));
    },
  };
};

const UpdateRace = withRouter(connect(
  mapStateToProps,
  mapDispatchToProps
)(PlayerRace));

export default UpdateRace;
