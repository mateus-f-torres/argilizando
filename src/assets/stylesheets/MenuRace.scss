
#RACE {}

/* for-tablet-landscape-up */
@media only screen and (min-width: 900px) {
  #RACE {
    .selection {
      grid-template-columns: repeat(3, auto);
      grid-template-rows: auto auto;
      grid-template-areas: "header header header" "secA secB secC";
      #title-box { grid-area: header; }
    }
    .card {
      grid-template-columns: 1fr 1fr;
      grid-template-rows: repeat(5, auto);
      grid-template-areas: "header header" "img stats" "traits traits" "rule rule" "btns btns";
      h2 { grid-area: header; }
      img { grid-area: img }
      #race-stats { 
        h3 { display: none; }
        grid-area: stats 
      }
      #race-traits { grid-area: traits }
      .lock-box { grid-area: btns }
    }
  }
}


