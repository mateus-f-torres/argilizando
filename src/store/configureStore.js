import {createStore} from 'redux';
import rootReducer from '../reducers/index.js';

export default function configureStore() {
  return createStore(rootReducer);
}
