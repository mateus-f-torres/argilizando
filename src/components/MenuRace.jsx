//@flow
import * as React from 'react';
import {Link, Route, Switch} from 'react-router-dom';
import CardRace from './CardRace.jsx';
import RACES from '../data/races.js';
import type {Race} from '../types/index.js';

import 'Styles/MenuRace.scss';

type Props = {
  getRace: (string) => void,
  lockRace: ({}) => void,
  race: Race,
  match: {
    url: string,
  },
};

class MenuRace extends React.Component<Props> {
  constructor(props: Props) {
    super(props);
    (this: any).showRace = this.showRace.bind(this);
    (this: any).lockRace = this.lockRace.bind(this);
  }

  showRace(e: SyntheticEvent<HTMLButtonElement>) {
    this.props.getRace(e.currentTarget.name);
  }

  lockRace() {
    this.props.lockRace(this.props.race);
  }

  render() {
    return (
      <section id="RACE">
        <Switch>

          <Route exact path="/race" render={()=>
            <ul className="selection">
              <div id="title-box">
                <h2>Player Race</h2>
                <div className="horizontal-line no-bottom" />
              </div>
              {
                RACES.map((category, i) => {
                  let [type, options] = [...category];

                  return (
                    <div key={i}>
                      <h3>{type}</h3>
                      <div className="horizontal-line no-bottom" />
                      <ul>
                        {
                          options.map((item) => (
                            <li key={item.id}>
                              <Link to={this.props.match.url + '/' + item.id}>
                                <button name={item.id} onClick={this.showRace}>
                                  {item.id}
                                </button>
                              </Link>
                            </li>
                          ))
                        }
                      </ul>
                    </div>
                  );
                })
              }
            </ul>
          }/>

          <Route path={this.props.match.url + '/'}
            render={()=>
              <CardRace lockRace={this.lockRace} {...this.props.race} />
            }/>
        </Switch>
      </section>
    );
  }
}

export default MenuRace;
