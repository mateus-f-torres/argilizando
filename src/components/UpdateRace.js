//@flow
import {connect} from 'react-redux';
import {withRouter} from 'react-router-dom';
import {displayRace} from 'Store/races/actions';
import {lockRace} from 'Store/done/actions';
import MenuRace from 'Components/UpdateRace/MenuRace.js';
import type {State} from 'Types/state.js';

const mapStateToProps = (state: State) => {
  return {
    race: state.race,
  };
};

const mapDispatchToProps = (dispatch: *) => {
  return {
    getRace: (race) => {
      dispatch(displayRace(race));
    },
    lockRace: (race) => {
      dispatch(lockRace(race));
    },
  };
};

const UpdateRace = withRouter(connect(
  mapStateToProps,
  mapDispatchToProps
)(MenuRace));

export default UpdateRace;
