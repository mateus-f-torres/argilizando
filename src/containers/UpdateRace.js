import {connect} from 'react-redux';
import {withRouter} from 'react-router-dom';
import {displayRace, lockRace} from '../actions';
import MenuRace from '../components/MenuRace.jsx';

const mapStateToProps = (state) => {
  return {
    race: state.race,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    getRace: (race) => {
      dispatch(displayRace(race));
    },
    lockRace: (race) => {
      dispatch(lockRace(race));
    },
  };
};

const UpdateRace = withRouter(connect(
  mapStateToProps,
  mapDispatchToProps
)(MenuRace));

export default UpdateRace;
