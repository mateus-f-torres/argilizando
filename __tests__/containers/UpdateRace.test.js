import React from 'react';
import {Provider} from 'react-redux';
import {MemoryRouter} from 'react-router-dom';

import configureStore from 'redux-mock-store';
import {mount} from 'enzyme';

import UpdateRace from 'Containers/UpdateRace.js';
import MenuRace from 'Components/MenuRace.jsx';

const mockState = {
  race: {},
};

const mockStore = configureStore();
const store = mockStore(mockState);

describe('UpdateRace', () => {
  beforeEach(() => {
    store.clearActions();
  });

  const container = mount(
    <Provider store={store}>
      <MemoryRouter initialEntries={[{pathname: '/race', key: 'testKey'}]}>
        <UpdateRace />
      </MemoryRouter>
    </Provider>
  );

  const component = container.find('MenuRace');

  // TODO check lockRace dispatch

  describe('Unit Test', () => {
    test('render MenuRace component', () => {
      expect(container.containsMatchingElement(<MenuRace />)).toBe(true);
    });

    test('receive store.race', () => {
      expect(component.prop('race')).toEqual(mockState.race);
    });

    test('dispatch getRace', () => {
      const humanBtn = component.find('button').filter('[name="human"]');

      const expectedActions = [
        {
          'type': 'DISPLAY_CHOSEN_RACE',
          'payload': 'human',
        },
      ];

      humanBtn.simulate('click');
      expect(store.getActions()).toEqual(expectedActions);
    });

    test('dispatch lockRace', () => {});
  });

  describe('Snapshot Test', () => {
    expect(container).toMatchSnapshot();
  });
});
