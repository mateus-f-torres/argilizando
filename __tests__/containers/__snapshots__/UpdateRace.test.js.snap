// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <MemoryRouter
    initialEntries={
      Array [
        Object {
          "key": "testKey",
          "pathname": "/race",
        },
      ]
    }
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "testKey",
              "pathname": "/race",
              "search": "",
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/race",
            "search": "",
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <withRouter(Connect(MenuRace))>
        <Route>
          <Connect(MenuRace)
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "canGo": [Function],
                "createHref": [Function],
                "entries": Array [
                  Object {
                    "hash": "",
                    "key": "testKey",
                    "pathname": "/race",
                    "search": "",
                  },
                ],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "index": 0,
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "key": "testKey",
                  "pathname": "/race",
                  "search": "",
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            location={
              Object {
                "hash": "",
                "key": "testKey",
                "pathname": "/race",
                "search": "",
              }
            }
            match={
              Object {
                "isExact": false,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
          >
            <MenuRace
              getRace={[Function]}
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "canGo": [Function],
                  "createHref": [Function],
                  "entries": Array [
                    Object {
                      "hash": "",
                      "key": "testKey",
                      "pathname": "/race",
                      "search": "",
                    },
                  ],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "index": 0,
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "key": "testKey",
                    "pathname": "/race",
                    "search": "",
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
              location={
                Object {
                  "hash": "",
                  "key": "testKey",
                  "pathname": "/race",
                  "search": "",
                }
              }
              lockRace={[Function]}
              match={
                Object {
                  "isExact": false,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
              race={Object {}}
            >
              <section
                id="RACE"
              >
                <Switch>
                  <Route
                    computedMatch={
                      Object {
                        "isExact": true,
                        "params": Object {},
                        "path": "/race",
                        "url": "/race",
                      }
                    }
                    exact={true}
                    location={
                      Object {
                        "hash": "",
                        "key": "testKey",
                        "pathname": "/race",
                        "search": "",
                      }
                    }
                    path="/race"
                    render={[Function]}
                  >
                    <ul
                      className="selection"
                    >
                      <div
                        id="title-box"
                      >
                        <h2>
                          Player Race
                        </h2>
                        <div
                          className="horizontal-line no-bottom"
                        />
                      </div>
                      <div
                        key="0"
                      >
                        <h3>
                          Common Races
                        </h3>
                        <div
                          className="horizontal-line no-bottom"
                        />
                        <ul>
                          <li
                            key="human"
                          >
                            <Link
                              replace={false}
                              to="//human"
                            >
                              <a
                                href="//human"
                                onClick={[Function]}
                              >
                                <button
                                  name="human"
                                  onClick={[Function]}
                                >
                                  human
                                </button>
                              </a>
                            </Link>
                          </li>
                          <li
                            key="halfling"
                          >
                            <Link
                              replace={false}
                              to="//halfling"
                            >
                              <a
                                href="//halfling"
                                onClick={[Function]}
                              >
                                <button
                                  name="halfling"
                                  onClick={[Function]}
                                >
                                  halfling
                                </button>
                              </a>
                            </Link>
                          </li>
                          <li
                            key="dwarf"
                          >
                            <Link
                              replace={false}
                              to="//dwarf"
                            >
                              <a
                                href="//dwarf"
                                onClick={[Function]}
                              >
                                <button
                                  name="dwarf"
                                  onClick={[Function]}
                                >
                                  dwarf
                                </button>
                              </a>
                            </Link>
                          </li>
                          <li
                            key="elf"
                          >
                            <Link
                              replace={false}
                              to="//elf"
                            >
                              <a
                                href="//elf"
                                onClick={[Function]}
                              >
                                <button
                                  name="elf"
                                  onClick={[Function]}
                                >
                                  elf
                                </button>
                              </a>
                            </Link>
                          </li>
                          <li
                            key="aasimar"
                          >
                            <Link
                              replace={false}
                              to="//aasimar"
                            >
                              <a
                                href="//aasimar"
                                onClick={[Function]}
                              >
                                <button
                                  name="aasimar"
                                  onClick={[Function]}
                                >
                                  aasimar
                                </button>
                              </a>
                            </Link>
                          </li>
                          <li
                            key="tiefling"
                          >
                            <Link
                              replace={false}
                              to="//tiefling"
                            >
                              <a
                                href="//tiefling"
                                onClick={[Function]}
                              >
                                <button
                                  name="tiefling"
                                  onClick={[Function]}
                                >
                                  tiefling
                                </button>
                              </a>
                            </Link>
                          </li>
                        </ul>
                      </div>
                      <div
                        key="1"
                      >
                        <h3>
                          Uncommon Races
                        </h3>
                        <div
                          className="horizontal-line no-bottom"
                        />
                        <ul>
                          <li
                            key="dragonborn"
                          >
                            <Link
                              replace={false}
                              to="//dragonborn"
                            >
                              <a
                                href="//dragonborn"
                                onClick={[Function]}
                              >
                                <button
                                  name="dragonborn"
                                  onClick={[Function]}
                                >
                                  dragonborn
                                </button>
                              </a>
                            </Link>
                          </li>
                          <li
                            key="goliath"
                          >
                            <Link
                              replace={false}
                              to="//goliath"
                            >
                              <a
                                href="//goliath"
                                onClick={[Function]}
                              >
                                <button
                                  name="goliath"
                                  onClick={[Function]}
                                >
                                  goliath
                                </button>
                              </a>
                            </Link>
                          </li>
                          <li
                            key="gnome"
                          >
                            <Link
                              replace={false}
                              to="//gnome"
                            >
                              <a
                                href="//gnome"
                                onClick={[Function]}
                              >
                                <button
                                  name="gnome"
                                  onClick={[Function]}
                                >
                                  gnome
                                </button>
                              </a>
                            </Link>
                          </li>
                          <li
                            key="kenku"
                          >
                            <Link
                              replace={false}
                              to="//kenku"
                            >
                              <a
                                href="//kenku"
                                onClick={[Function]}
                              >
                                <button
                                  name="kenku"
                                  onClick={[Function]}
                                >
                                  kenku
                                </button>
                              </a>
                            </Link>
                          </li>
                          <li
                            key="lizardfolk"
                          >
                            <Link
                              replace={false}
                              to="//lizardfolk"
                            >
                              <a
                                href="//lizardfolk"
                                onClick={[Function]}
                              >
                                <button
                                  name="lizardfolk"
                                  onClick={[Function]}
                                >
                                  lizardfolk
                                </button>
                              </a>
                            </Link>
                          </li>
                          <li
                            key="tabaxi"
                          >
                            <Link
                              replace={false}
                              to="//tabaxi"
                            >
                              <a
                                href="//tabaxi"
                                onClick={[Function]}
                              >
                                <button
                                  name="tabaxi"
                                  onClick={[Function]}
                                >
                                  tabaxi
                                </button>
                              </a>
                            </Link>
                          </li>
                        </ul>
                      </div>
                      <div
                        key="2"
                      >
                        <h3>
                          Monstrous Races
                        </h3>
                        <div
                          className="horizontal-line no-bottom"
                        />
                        <ul>
                          <li
                            key="goblin"
                          >
                            <Link
                              replace={false}
                              to="//goblin"
                            >
                              <a
                                href="//goblin"
                                onClick={[Function]}
                              >
                                <button
                                  name="goblin"
                                  onClick={[Function]}
                                >
                                  goblin
                                </button>
                              </a>
                            </Link>
                          </li>
                          <li
                            key="orc"
                          >
                            <Link
                              replace={false}
                              to="//orc"
                            >
                              <a
                                href="//orc"
                                onClick={[Function]}
                              >
                                <button
                                  name="orc"
                                  onClick={[Function]}
                                >
                                  orc
                                </button>
                              </a>
                            </Link>
                          </li>
                          <li
                            key="bugbear"
                          >
                            <Link
                              replace={false}
                              to="//bugbear"
                            >
                              <a
                                href="//bugbear"
                                onClick={[Function]}
                              >
                                <button
                                  name="bugbear"
                                  onClick={[Function]}
                                >
                                  bugbear
                                </button>
                              </a>
                            </Link>
                          </li>
                          <li
                            key="hobgoblin"
                          >
                            <Link
                              replace={false}
                              to="//hobgoblin"
                            >
                              <a
                                href="//hobgoblin"
                                onClick={[Function]}
                              >
                                <button
                                  name="hobgoblin"
                                  onClick={[Function]}
                                >
                                  hobgoblin
                                </button>
                              </a>
                            </Link>
                          </li>
                          <li
                            key="kobold"
                          >
                            <Link
                              replace={false}
                              to="//kobold"
                            >
                              <a
                                href="//kobold"
                                onClick={[Function]}
                              >
                                <button
                                  name="kobold"
                                  onClick={[Function]}
                                >
                                  kobold
                                </button>
                              </a>
                            </Link>
                          </li>
                          <li
                            key="yuanti"
                          >
                            <Link
                              replace={false}
                              to="//yuanti"
                            >
                              <a
                                href="//yuanti"
                                onClick={[Function]}
                              >
                                <button
                                  name="yuanti"
                                  onClick={[Function]}
                                >
                                  yuanti
                                </button>
                              </a>
                            </Link>
                          </li>
                        </ul>
                      </div>
                    </ul>
                  </Route>
                </Switch>
              </section>
            </MenuRace>
          </Connect(MenuRace)>
        </Route>
      </withRouter(Connect(MenuRace))>
    </Router>
  </MemoryRouter>
</Provider>
`;
